### ISSUE TITLE
List of unimplemented abilities and number of affected buttons

### ISSUE THREAD

**Issue by TheOrgg** [buttonmen-dev/buttonmen#2620]
I was doing this for myself, but thought it might be a good reference for planning out where to go when Skills get focused on again.  I'll start with the two BSS has said are his priority.  These are roughly in order of how many buttons would be affected; if I didn't remember any of the other obscure skills, please add them here.  I listed the "Fudged Dice" at the end.  I felt this was a better place for this than the wiki, as I hope all of these are at some point put into effect here, though a few I think need playtesting to make sure they function properly, like Teleport Dice.

Chaotic Dice (@)
Hodgepodge (3 buttons)
States (7 buttons)
Fanatics (8 buttons)

Wildcard Dice ("(C)")
Las Vegas (1 button)
Fanatics (7)

Select Dice
Fightball (12 buttons)
High School Drama (8 buttons) (rS dice)
Ice Cream Warriors (9 buttons) [untested set]

Lucky Dice
Murderers (17 buttons)

Rebound Dice
Bridge and Tunnel (7 buttons)
MLP (1 button)

Dice Batches
Monstrous Musicians (8 buttons)

Gamer Dice
Origins (3 buttons)
MLP (2 buttons) 

Armor Dice
Nodwick (5 buttons)

Assassin Dice
Order of Dolls (4 buttons)

Plasma Dice
Space Girlz (2 buttons)
Gaming Guardians (1 button)

Teleport Dice
Gaming Guardians (2 buttons)
MLP (1 button)

Triplet Dice
Las Vegas (1 button)
Warlords (2 buttons)

Evil Dice
Gaming Guardians (2 buttons)
Hodgepodge (1 button)

Aggressive Dice
Blademasters 2/3 (2 buttons)

Cross Dice
Blademasters 2/3 (2 buttons)

Defensive Dice
Blademasters 2/3 (2 buttons)

Thief Dice
L5R: Everything 2 Gain (2 buttons)

Full Auto
MegaTokyo (1 button)

Dual Dice
Las Vegas (1 button)

Deception Dice
Gaming Guardians (1 button)

Kubla's Treasure Dice
Kublacon (1 button)

Hodgekins and Chase's Giant-Sized Dice
Kublacon (1 button)

Not a skill, but is what is.

Uniquely numbered or "Fudged" dice
Las Vegas (1 button)
Blademasters (1 button)
Cheese Weasel (1 button)
Formula De (?) (8 buttons?  Research is ongoing)

**Comment by blackshadowshade** [comment]
In addition to this, I've also got button skills on my horizon. First of those will be Socrates, then I'll take a look at the iconic abilities for Sailor Moon to see if there are any easy wins there.

**Comment by jl8e** [comment]
Cheese weasel doesn't have a weirdly numbered die

**Comment by TheOrgg** [comment]
Yeah, there's a few button specials that aren't active yet, too.  Cheese Weasel comes to mind.

Oops, Buzzing Weasel, not Cheese Weasel.  Cheese Weasel has a Button Special Ability that hopefully won't be too hard.  I also forgot to check Fanatics for Plasma.

Personally, I'm hoping Select and Triplets are coming soon-ish.  

When we test Select, should we also playtest Ice Cream Warriors?  They were on the original site almost no time, as Select didn't work right.  We've got permission for doing so.

**Comment by jl8e** [comment]
Buzzing weasel is not exactly a compelling priority. (Which isn't to say that non-linear distribution dice aren't,  but BW isn't a good argument for them.)

(Also, it's not a "Fudged" die. There is no general term. The Fudge die was created for the RPG FUDGE, thus the name for the even-distribution  (1, 0, -1) die.)

**Comment by jl8e** [comment]
Also, Triplets: easy, mostly UI work. On the other hand, we have exactly one usable button recipe that needs them, and until last week we had zero, so don't expect anyone to bother.

Select: not easy

**Comment by TheOrgg** [comment]
If it's easy, why not just get it done?  (And the final form of Vegas has a Triplet-- (10,10,1))

I was unaware there was an RPG system using such a die.  My bad.  

I realize most of these aren't easy.  I tried to learn php, but just ended up getting lost-- you need a certain kind of mind to program, and I just don't have it.  I'm slowly figuring out image editing.

**Comment by cgolubi1** [comment]
I don't believe triplets are going to be easy, and i think my understanding of the lift based on the state of the codebase and the kinds of things that are likely to go wrong (logic bugs, and brittle assumptions in many different parts of the code, mostly), is probably as accurate as anyone's with the possible/likely exception of Shadowshade.

**Comment by jl8e** [comment]
Because even "easy" is still work. Even if it's exactly as easy on the back end as changing the two-die cap on twin dice (and it likely isn't, because there are always details), it still needs to be written,  automated tests need writing, it needs playtesting,  and then the front end needs updating to handle displaying them properly. Which, again, may not be hard, but triple-bubbles may well cause layout issues, particularly once some jerk decides to make an all-triple fanatic. (And you know they will, so you have to anticipate it now.)

And, of course, any skill needs testing for interactions with other skills. You may think that triple would be easy because it's just more twin, but not checking is how you end up regretting your life choices. (Obvious potential problems: fire, stinger, and _radioactive_)

Even if it could be knocked out in an afternoon for each, it's still an afternoon they're not spending on other things.

**Comment by blackshadowshade** [comment]
Just for perspective, I had to create a wiki page to remind myself how to add a new die skill in code: https://github.com/buttonmen-dev/buttonmen/wiki/developer-guide%3A-adding-die-skills

There is no "simply add a simple die skill in a day" unless I devote a solid number of hours to it.

**Comment by blackshadowshade** [comment]
Actually, following up on my previous post, the *major* reason why we don't do loads of things at the same time is that they will conflict, both physically (as in, I have to deconflict pull requests) and logically (skills may interact in odd ways that we can't pick up in testing until they're both present).

Our major blockers for further development on skills at the moment are CustomBM and tournaments. These pull requests touch so many files that unintended consequences are almost a given. And if you've been following the CustomBM pull request, we're currently trying to stop site-breaking skill combinations or die types from occurring in custom recipes.
